File: EmojiPanel/src/main/java/com/binlee/emoji/model/DbManager.java
package com.binlee.emoji.model;

import java.util.List;

public final class DbManager {

    private DbManager() {
    }

    public void updateLastUsedEmojiGroup(List<EmojiGroup> groups, int uid) {
    }

    public List<Emoji> queryRecentUsedEmojis(int uid) {
        return null;
    }

    public void updateRecentUsedEmojis(List<Emoji> emojis, int uid) {
    }

    public List<Emoji> queryEmojiListByGroupUuid(String uuid, int uid) {
        return null;
    }

    public List<Emoji> queryCustomEmojis(int uid) {
        return null;
    }

    public boolean hasEmojiGroup(String uuid) {
        return false;
    }

    public boolean hasCustomEmoji(String sha1) {
        return false;
    }

    public EmojiGroup getCustomEmojiGroup(int userId) {
        return null;
    }

    public EmojiGroup queryEmojiGroupByType(int type, int userId) {
        return null;
    }

    public void insertEmojiGroup(EmojiGroup group, int userId) {
    }

    public void insertEmojiList(List<Emoji> list, int userId) {
    }

    public List<EmojiGroup> queryAllEmojiGroup(int userId) {
        return null;
    }

    public void deleteEmojisByGroup(String uuid) {
    }

    public void updateEmojiGroupNameThumbnailCount(EmojiGroup group, int userId) {
    }

    public EmojiGroup queryEmojiGroupByUuid(String uuid, int userId) {
        return null;
    }

    public void deleteEmojiList(int userId, List<Emoji> list) {
    }

    private static class Singleton {
        static DbManager sManager = new DbManager();
    }

    public static DbManager getInstance() {
        return Singleton.sManager;
    }
}

File: EmojiPanel/src/main/java/com/binlee/emoji/model/DbManager.kt
package com.binlee.emoji.model

import android.util.Log

class DbManager private constructor() {

    companion object {
        fun get(): DbManager = Singleton.manager
        private const val TAG = "DbManager"
    }

    fun updateLastUsedEmojiGroup(groups: List<EmojiGroup?>?, uid: Int) {
        Log.d(TAG, "updateLastUsedEmojiGroup() called with: groups = $groups, uid = $uid")
    }

    fun queryRecentUsedEmojis(uid: Int): List<Emoji>? {
        Log.d(TAG, "queryRecentUsedEmojis() called with: uid = $uid")
        return null
    }
    fun updateRecentUsedEmojis(emojis: List<Emoji?>?, uid: Int) {
        Log.d(TAG, "updateRecentUsedEmojis() called with: emojis = $emojis, uid = $uid")
    }

    fun queryEmojiListByGroupUuid(uuid: String?, uid: Int): List<Emoji>? {
        Log.d(TAG, "queryEmojiListByGroupUuid() called with: uuid = $uuid, uid = $uid")
        return null
    }

    fun queryCustomEmojis(uid: Int): List<Emoji>? {
        Log.d(TAG, "queryCustomEmojis() called with: uid = $uid")
        return null
    }

    fun hasEmojiGroup(uuid: String?): Boolean {
        Log.d(TAG, "hasEmojiGroup() called with: uuid = $uuid")
        return false
    }

    fun hasCustomEmoji(sha1: String?): Boolean {
        Log.d(TAG, "hasCustomEmoji() called with: sha1 = $sha1")
        return false
    }

    fun getCustomEmojiGroup(userId: Int): EmojiGroup? {
        Log.d(TAG, "getCustomEmojiGroup() called with: userId = $userId")
        return null
    }

    fun queryEmojiGroupByType(type: Int, userId: Int): EmojiGroup? {
        Log.d(TAG, "queryEmojiGroupByType() called with: type = $type, userId = $userId")
        return null
    }

    fun insertEmojiGroup(group: EmojiGroup?, userId: Int) {
        Log.d(TAG, "insertEmojiGroup() called with: group = $group, userId = $userId")
    }

    fun insertEmojiList(list: List<Emoji>?, userId: Int) {
        Log.d(TAG, "insertEmojiList() called with: list = $list, userId = $userId")
    }

    fun queryAllEmojiGroup(userId: Int): List<EmojiGroup>? {
        Log.d(TAG, "queryAllEmojiGroup() called with: userId = $userId")
        return null
    }

    fun deleteEmojisByGroup(uuid: String?) {
        Log.d(TAG, "deleteEmojisByGroup() called with: uuid = $uuid")
    }

    fun updateEmojiGroupNameThumbnailCount(group: EmojiGroup?, userId: Int) {
        Log.d(TAG, "updateEmojiGroupNameThumbnailCount() called with: group = $group, userId = $userId")
    }

    fun queryEmojiGroupByUuid(uuid: String?, userId: Int): EmojiGroup? {
        Log.d(TAG, "queryEmojiGroupByUuid() called with: uuid = $uuid, userId = $userId")
        return null
    }

    fun deleteEmojiList(userId: Int, list: List<Emoji>?) {
        Log.d(TAG, "deleteEmojiList() called with: userId = $userId, list = $list")
    }

    private object Singleton {
        val manager = DbManager()
    }
}